### PR TITLE
chore: add examples of using esm with jest_test both .mjs files and with 'type':'module' in package.json

### PR DESCRIPTION
**Pull request by gregmagolan**
Resolves https://github.com/aspect-build/rules_jest/issues/65

### CODE REVIEW

File: example/sharding/index.test.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const indexfile = require("./index.js");
```
Comment (jbedard) [inline example/sharding/index.test.js:1]: Do we want this change in a `sharding` test? Rename the file instead? Or don't change this example? Doesn't really matter though 🤷 

Interesting that this had the `.js` extension in the require statement 🤔

**Review by jbedard — APPROVED** [review]

File: example/sharding/index.test.js (hunk 1 above)
Comment (gregmagolan) [reply example/sharding/index.test.js:1]: It's just redundant in a require statement so may as well make our examples clean without redundancies. Node will look for an index.js or index.mjs or index.cjs if you require a folder.

Comment (jbedard) [reply example/sharding/index.test.js:1]: Yeah I don't think we should ever have the file extension in there.

But an index is normally to group a folder so the folder can be imported while hiding how many files it has in it, not a random file like this for importing directly. Doesn't really relate to this PR though 🤷

Comment (gregmagolan) [reply example/sharding/index.test.js:1]: Yeah. Suppose it is a contrived example in that way.

**Pull request merged**